src/topology/order/priestley.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Sara Rousta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Sara Rousta
+-/
+import order.upper_lower
+import topology.separation
+
+/-!
+# Priestley spaces
+
+This file defines Priestley spaces. A Priestley space is an ordered compact topological space such
+that any two distinct points can be separated by a clopen upper set.
+
+## Main declarations
+
+* `priestley_space`: Prop-valued mixin stating the Priestley separation axiom: Any two distinct
+  points can be separated by a clopen upper set.
+
+## Implementation notes
+
+We do not include compactness in the definition, so a Priestley space is to be declared as follows:
+`[preorder α] [topological_space α] [compact_space α] [priestley_space α]`
+
+## References
+
+* [Wikipedia, *Priestley space*](https://en.wikipedia.org/wiki/Priestley_space)
+* [Davey, Priestley *Introduction to Lattices and Order*][davey_priestley]
+-/
+
+open set
+
+variables {α : Type*}
+
+/-- A Priestley space is an ordered topological space such that any two distinct points can be
+separated by a clopen upper set. Compactness is often assumed, but we do not include it here. -/
+class priestley_space (α : Type*) [preorder α] [topological_space α] :=
```

Comment:
Subclassing `topological_space` and `preorder` by instance arguments, like you do here, matches [`order_closed_topology`](https://leanprover-community.github.io/mathlib_docs/find/order_closed_topology) and is probably the best idea. I have a vague idea crystallizing in my head that separating data-classes and prop-classes makes the system a lot more flexible, for example switching out the exact kind of relation in ordered rings, etc.